### PR TITLE
ADBDEV-6011: Fix restore helper hides actual err

### PR DESCRIPTION
**Pull request by RekGRpth**
Fix restore helper hides actual err

The commit 4371a94 moved the LoopEnd label so that the err variable began to be
overwritten and it will no longer be used below. To fix it, this patch renames
the err variable returned by flushAndCloseRestoreWriter to errPipe.

The integration test has been adapted to error-free behavior.

---
Fix restore helper hides actual err

The commit 4371a94 moved the LoopEnd label so that the err https://github.com/arenadata/gpbackup/blob/9b07afe59f42d7aee57e6e986f249198547d77cf/helper/restore_helper.go#L308-L310 error began to be overwritten https://github.com/arenadata/gpbackup/blob/9b07afe59f42d7aee57e6e986f249198547d77cf/helper/restore_helper.go#L322 and it will no longer be used below https://github.com/arenadata/gpbackup/blob/9b07afe59f42d7aee57e6e986f249198547d77cf/helper/restore_helper.go#L335
To fix it, this patch renames err variable returned by flushAndCloseRestoreWriter to errPipe.

### CODE REVIEW

**Comment by dkovalev1** [comment]
Looks logical, but what is the scenario when it does matter? Can we reproduce the problem fixed and make a test?

**Comment by RekGRpth** [comment]
> Looks logical, but what is the scenario when it does matter? Can we reproduce the problem fixed and make a test?

Problem was found theoretically

**Comment by red1452** [comment]
Change the commit message:
```
The commit https://github.com/arenadata/gpbackup/commit/4371a94305a7a77ce3e6990a9691a3986d9426d9 moved the LoopEnd label so that the err variable began to be
overwritten and it will no longer be used below. To fix it, this patch renames
the err variable returned by flushAndCloseRestoreWriter to errPipe.
```

**Comment by RekGRpth** [comment]
> Change the commit message:
> 
> ```
> The commit https://github.com/arenadata/gpbackup/commit/4371a94305a7a77ce3e6990a9691a3986d9426d9 moved the LoopEnd label so that the err variable began to be
> overwritten and it will no longer be used below. To fix it, this patch renames
> the err variable returned by flushAndCloseRestoreWriter to errPipe.
> ```

changed

**Review by red1452 — DISMISSED** [review]

**Comment by dkovalev1** [comment]
> > Looks logical, but what is the scenario when it does matter? Can we reproduce the problem fixed and make a test?
> 
> Problem was found theoretically

Can you at least explain when the problem happens and which cases to patch solves? If it's not possible to supply a test.

**Comment by RekGRpth** [comment]
> > > Looks logical, but what is the scenario when it does matter? Can we reproduce the problem fixed and make a test?
> > 
> > 
> > Problem was found theoretically
> 
> Can you at least explain when the problem happens and which cases to patch solves? If it's not possible to supply a test.

test was added

File: end_to_end/end_to_end_suite_test.go [hunk 1]
```diff
@@ -2519,6 +2519,19 @@ LANGUAGE plpgsql NO SQL;`)
 			Expect(string(output)).To(MatchRegexp(`Error loading data into table public.d: COPY d, line 1: "\d,d": ERROR: null value in column "c" violates not-null constraint`))
 			Expect(string(output)).To(MatchRegexp(`Error loading data into table public.e: COPY e, line 1: "\d,e": ERROR: null value in column "c" violates not-null constraint`))
 			Expect(string(output)).To(ContainSubstring(`Encountered 3 error(s) during table data restore`))
+			homeDir := os.Getenv("HOME")
+			helperFiles, _ := path.Glob(path.Join(homeDir, "gpAdminLogs/gpbackup_helper_*"))
+			helperCommand := exec.Command("grep", "Error copying data", helperFiles[len(helperFiles)-1])
+			helperOutput, _ := helperCommand.CombinedOutput()
+			Expect(string(helperOutput)).To(ContainSubstring(`Segment 2: Oid 16397, Batch 0: Error encountered: Error copying data: write`))
```
Comment (red1452) [inline end_to_end/end_to_end_suite_test.go:2526]: Can you remove oid from these lines?

Comment (RekGRpth) [reply end_to_end/end_to_end_suite_test.go:2526]: oid is taken from backup and always same

**Review by dkovalev1 — APPROVED** [review]
This change seems logical and allows gpbackup to report error. However, the test system is far from parfect. Tests are very difficult to read, trace and understand.

**Review by red1452 — APPROVED** [review]

**Pull request merged**